src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5938,6 +5960,19 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             }
 
             close = this.EatToken(SyntaxKind.GreaterThanToken);
+
+            static bool tokenKindBreaksArgumentList(SyntaxKind kind)
+            {
+                return kind
+                    is SyntaxKind.LessThanToken
+                    or SyntaxKind.OpenParenToken
+                    or SyntaxKind.CloseParenToken
+                    or SyntaxKind.SemicolonToken
+                    or SyntaxKind.OpenBraceToken
+                    or SyntaxKind.CloseBraceToken
+                    or SyntaxKind.EqualsToken
+                    or SyntaxKind.EqualsGreaterThanToken;
```

Comment:
It would be nice to see examples of all of the above.

Reply:
Are there tests for each of these cases?

Reply:
I think some of those might be missing some tests, will make another pass

Reply:
this comment still applies.  i want to know why each of those cases are something that would break the argument list.

Reply:
comment still applies. :)